GAE/src/org/akvo/flow/rest/ReportRestService.java

```diff
@@ -0,0 +1,217 @@
+                // Save it, so we get an id assigned
+                r = reportDao.save(r);
+
+                //Queue it
+                Queue queue = QueueFactory.getDefaultQueue();
+                TaskOptions options = TaskOptions.Builder.withUrl("/app_worker/reportservlet")
+                        .param(TaskRequest.ACTION_PARAM, ReportTaskRequest.START_ACTION)
+                        .param(ReportTaskRequest.ID_PARAM, Long.toString(r.getKey().getId()));
+                queue.add(options); //overwrite any supplied state
+                dto = new ReportDto();
+                DtoMarshaller.copyToDto(r, dto);
+                statusDto.setStatus("ok");
+            }
+        }
+
+        response.put("meta", statusDto);
+        response.put("report", dto);
+        return response;
+    }
+
+    // find all reports belonging to the current user
+    //TODO: get an unfiltered list if superAdmin?
+    @RequestMapping(method = RequestMethod.GET, value = "")
+    @ResponseBody
+    public Map<String, Object> listMyReports() {
+        final Map<String, Object> response = new HashMap<String, Object>();
+        List<ReportDto> results = new ArrayList<ReportDto>();
+        List<Report> reports = reportDao.listAllByCurrentUser();
+        if (reports != null) {
+            for (Report r : reports) {
+                ReportDto dto = new ReportDto();
+                DtoMarshaller.copyToDto(r, dto);
+                results.add(dto);
+            }
+        }
+        response.put("reports", results);
+        return response;
+    }
+
+    // find a single report by the reportId
+    // TODO: restrict use to owner+superAdmins
+    @RequestMapping(method = RequestMethod.GET, value = "/{id}")
+    @ResponseBody
+    public Map<String, ReportDto> findReport(@PathVariable("id") Long id) {
+        final Map<String, ReportDto> response = new HashMap<String, ReportDto>();
+        Report qo = reportDao.getByKey(id);
+        ReportDto dto = null;
+        if (qo != null) {
+            dto = new ReportDto();
+            DtoMarshaller.copyToDto(qo, dto);
+        }
+        response.put("report", dto);
+        return response;
+
+    }
+
+    // delete report by id
+    @RequestMapping(method = RequestMethod.DELETE, value = "/{id}")
+    @ResponseBody
+    public Map<String, RestStatusDto> deleteReportById(@PathVariable("id") Long id) {
+        final Map<String, RestStatusDto> response = new HashMap<String, RestStatusDto>();
+        Report qo = reportDao.getByKey(id);
+        RestStatusDto statusDto = null;
+        statusDto = new RestStatusDto();
+        statusDto.setStatus("failed");
+
+        // check if report exists in the datastore
+        if (qo != null) {
+            // delete report group
+            reportDao.delete(qo);
+            statusDto.setStatus("ok");
+        }
+        response.put("meta", statusDto);
+        return response;
+    }
+
+    // update existing report
+    //TODO: only allow status changes?
```

Comment:
I think this makes sense.  Only update the status 👍

Reply:
If we are moving the updates from the report engine to the "servlet" API then no changes will be needed through this API at all. PUT can be completely removed.

Reply:
Sure. if its not needed then we get rid of it till we actually need it.